### PR TITLE
tracking the memory usage of orc-booting compiler for each commit

### PR DESCRIPTION
**Pull request by ringabout**
I'm testing an action bot which will publish the compilation time and memory usage for each commit/merge.

### CODE REVIEW

File: ci/action.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+const GITHUB_TOKEN = core.getInput('GITHUB_TOKEN');
+const octokit = github.getOctokit(GITHUB_TOKEN);
+
+const { context = {} } = github;
+const { pull_request } = context.payload;
+
+await octokit.issues.createComment({
+  ...context.repo,
+  issue_number: pull_request.number,
+  body: `Thank you for submitting a pull request! We will try to review this as soon as we can.\n\n<img src="${gifUrl}" alt="thank you" />`
```
Comment (juancarlospaco) [inline ci/action.js]: This already exists, you dont have to maintain it.
- https://github.com/actions/first-interaction#usage

**Comment by demotomohiro** [comment]
Does it measure the compilation time in same CPU and storage? If the compilation times is not measured in the same condition, it is not much useful.

**Comment by ringabout** [comment]
I agree. We need self hosted github action eventually => https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners

File: ci/action.js (hunk 1 above)
Comment (ringabout) [reply ci/action.js]: Yeah, that is a demo for a start.

**Comment by ringabout** [comment]
> Does it measure the compilation time in same CPU and storage? If the compilation times is not measured in the same condition, it is not much useful.

As a workaround for now, we can benchmark compiler without this change and compiler with this change at the same machine.

**Comment by juancarlospaco** [comment]
Just steal https://github.com/rust-lang/rust/blob/master/src/ci/cpu-usage-over-time.py

**Comment by ringabout** [comment]
The memory usage seems to be reliable even using Github Action.

**Comment by ringabout** [comment]
We could probably use valgrind to profile the Nim compiler and send the result (.out and a png) to the comment.

**Comment by Araq** [comment]
> We could probably use valgrind to profile the Nim compiler and send the result (.out and a png) to the comment.

But not for every CI run, valgrind can be really slow.

**Comment by Araq** [comment]
@xflywind if you need it, feel free to merge it but at this point I have no idea anymore how our CI works...

**Pull request merged**